lib/ansible/modules/cloud/azure/azure_rm_aci.py

```diff
@@ -0,0 +1,404 @@
+    :param: aci: Container
+    :return: dict with the state on Azure
+    '''
+    results = dict(
+        id=aci.id,
+        name=aci.name,
+        tags=aci.tags,
+        location=aci.location,
+        type=aci.type,
+        ip_address=aci.ip_address,
+        restart_policy=aci.restart_policy,
+        state=aci.state,
+        provisioning_state=aci.provisioning_state,
+        image_registry_credentials=aci.image_registry_credentials,
+        volumes=aci.volumes,
+        os_type=aci.os_type
+    )
+    return results
+
+
+class AzureRMContainerInstance(AzureRMModuleBase):
+    """Configuration class for an Azure RM container instance resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            group_name=dict(
+                type='str',
+                required=True
+            ),
+            image=dict(
+                type='str',
+                required=True
+            ),
+            memory=dict(
+                type='float',
+                required=False,
+                default=1.5
+            ),
+            cpu=dict(
+                type='int',
+                required=False,
+                default=1
+            ),
+            os_type=dict(
+                type='str',
+                required=False,
+                default='linux',
+                choices=['linux', 'windows']
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            ip_address=dict(
+                type='str',
+                required=False,
+                default=None,
+                choices=['public']
+            ),
+            port=dict(
+                type='int',
+                required=False,
+                default=80
+            ),
+            registry_login_server=dict(
+                type='str',
+                required=False,
+                default=None
+            ),
+            registry_username=dict(
+                type='str',
+                required=False,
+                default=None
+            ),
+            registry_password=dict(
+                type='str',
+                required=False,
+                default=None
+            ),
+            service_principal=dict(
+                type='list',
+                required=False
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.tags = None
+        self.state = None
+        self.service_principal = None
```

Comment:
These members seems not used here. I guess they are from container service module. You may want to clean these up.